services/rfq/api/client/client.go

```diff
@@ -50,12 +58,14 @@ func (c unauthenticatedClient) resty() *resty.Client {
 
 type clientImpl struct {
 	UnauthenticatedClient
-	rClient *resty.Client
+	rClient   *resty.Client
+	wsURL     *string
+	reqSigner signer.Signer
 }
 
 // NewAuthenticatedClient creates a new client for the RFQ quoting API.
 // TODO: @aurelius,  you don't actually need to be authed for GET Requests.
-func NewAuthenticatedClient(metrics metrics.Handler, rfqURL string, reqSigner signer.Signer) (AuthenticatedClient, error) {
+func NewAuthenticatedClient(metrics metrics.Handler, rfqURL string, wsURL *string, reqSigner signer.Signer) (AuthenticatedClient, error) {
```

Comment:
Yeah that was my thinking, not opposed to making it required just thought it might be convenient to allow running without websocket server. IMO overkill to use option pattern for one param but if we had multiple here I would be in favor of it